### ISSUE TITLE
Use GetPathOfFileAbove where appropriate

### ISSUE THREAD

**Issue by ivdiazsa** [dotnet/corefx#39987]
Following up from CoreCLR's repo Issue 20582, there are certain places in the source code where GetPathOfFileAbove usage is preferred to the more verbose GetDirectoryNameOfFileAbove. The following example illustrates both calls:

Change:
- `"$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props"`

To:
- `"$([MSBuild]::GetPathOfFileAbove(dir.props))"`

The second one is easier to read and less prone to mistakes if edition is ever required. Are there any reasons to not use GetPathOfFileAbove in CoreFX?

**Comment by ivdiazsa** [comment]
Adding @danmosemsft and @safern for updates.

**Comment by danmosemsft** [comment]
If it's functionally the same, and it works where we need it to, it should  be fine.

**Comment by safern** [comment]
This might not work based on: https://github.com/dotnet/coreclr/pull/25951#issuecomment-518413714

This is because in corefx we mostly do a recursive import of `Directory.Build.props` from within files with the same name, just to confirm if that limitation for `GetPathOfFileAbove` is by design or not.

**Comment by ivdiazsa** [comment]
Taking from the discussion in dotnet/coreclr#25951:

> I think he meant, if you're in a file called foo.props and you're trying to get a file from a directory above recursively that has the same name, in this example foo.props. So he's saying that using GetPathOfFileAbove(Directory.Build.props) won't work in files named Directory.Build.props.

@safern Let me see if I understood this correctly. You can't use `GetPathOfFileAbove(Directory.Build.props)` within a file called `Directory.Build.props`, right?

I ran a search throughout the CoreFX `src` directory and subdirectories and found several uses of `GetDirectoryNameOfFileAbove(Directory.Build.props,...)` under files with different names, mainly pkgproj files. Would these files benefit from the change to `GetPathOfFileAbove(Directory.Build.props)` or would this make the build fail should the `Directory.Build.props` file be located somewhere above in the same directory tree?

**Comment by ViktorHofer** [comment]
Yes anything that imports a file "from above" which hasn't the same name as the calling msbuild file is suitable.
